torch/_functorch/_aot_autograd/runtime_wrappers.py

```diff
@@ -147,6 +148,104 @@ def post_compile(
         )
 
 
+class NoopAliasHandler:
+    def __init__(self, info, runtime_metadata, trace_joint):
+        pass
+
+    def __call__(self, orig_inputs, fw_outs, out):
+        return out
+
+
+def _unwrap_tensoralias(x):
+    assert isinstance(x, TensorAlias)
+    return x.alias
+
+
+def _identity(x):
+    return x
+
+
+class AliasOfInputHandler:
+    def __init__(self, info, runtime_metadata, trace_joint):
+        num_tokens = len(runtime_metadata.tokens)
+        self.base_idx = info.base_idx + num_tokens
+        self.unwrap_out = _unwrap_tensoralias if trace_joint else _identity
+        self.requires_grad = info.requires_grad
+        self.functional_tensor = info.functional_tensor
+        self.replay_views = config.view_replay_for_aliased_outputs
+
+    def __call__(self, orig_inputs, fw_outs, out):
+        aliased_base_tensor = orig_inputs[self.base_idx]
+        return gen_alias_from_base(
+            aliased_base_tensor,
+            self.unwrap_out(out),
+            self.requires_grad,
+            self.functional_tensor,
+            replay_views=self.replay_views,
+        )
+
+
+class IsInputHandler:
+    def __init__(self, info, runtime_metadata, trace_joint):
+        num_tokens = len(runtime_metadata.tokens)
+        self.base_idx = info.base_idx + num_tokens
+        self.unwrap_out = _unwrap_tensoralias if trace_joint else _identity
+
+    def __call__(self, orig_inputs, fw_outs, out):
+        aliased_base_tensor = orig_inputs[self.base_idx]
+        return aliased_base_tensor
+
+
+class AliasOfIntermediateHandler:
+    def __init__(self, info, runtime_metadata, trace_joint):
+        if info.output_type in (
+            OutputType.alias_of_intermediate,
+            OutputType.alias_of_intermediate_save_as_output,
+        ):
+            num_user_outputs = len(runtime_metadata.output_info)
+            self.base_idx = info.base_idx + num_user_outputs
+        else:
+            self.base_idx = info.base_idx
+
+        self.unwrap_out = _unwrap_tensoralias if trace_joint else _identity
+        self.requires_grad = info.requires_grad
+        self.functional_tensor = info.functional_tensor
+        self.replay_views = config.view_replay_for_aliased_outputs
+
+    def __call__(self, orig_inputs, fw_outs, out):
+        aliased_base_tensor = fw_outs[self.base_idx]
+        return gen_alias_from_base(
+            aliased_base_tensor,
+            self.unwrap_out(out),
+            self.requires_grad,
+            self.functional_tensor,
+            replay_views=self.replay_views,
+        )
+
+
+def make_output_handler(info, runtime_metadata, trace_joint):
```

Comment:
@jamesjwu recently added a warm cache on top of AOTAutograd that involves serializing all metadata that we need at runtime in the cache (and the existing `TestAOTAutograd` tests won't exercise the cache until https://github.com/pytorch/pytorch/pull/128222 lands).

I don't think these changes will affect the cache though, since we shouldn't actually need to serialize / save any of these new runtime handlers. The ordering goes:

(1) on cache hit: load the correct inductor cache key, and the `ViewAndMutationMeta` (runtime metadata for AOTAutograd)

(2) invoke the [post_compile](https://github.com/pytorch/pytorch/blob/main/torch/_functorch/_aot_autograd/autograd_cache.py#L342) of each wrapper layer in AOTAutograd, one of which will call `_create_runtime_wrapper`

(3) then (still at compile, but after the cache lookup finishes), we will create these handlers to avoid some runtime overhead.

Reply:
Yup, the nice thing about the refactors we've done is that as long as the API of _create_runtime_wrapper and RuntimeWrapper don't change, you're probably not introducing any new state that needs caching